src/main/java/com/ericsson/gerrit/plugins/eiffel/handlers/DataBaseHandler.java

```diff
@@ -0,0 +1,233 @@
+package com.ericsson.gerrit.plugins.eiffel.handlers;
+
+import java.io.File;
+import java.net.ConnectException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is the database handler class. at initiation it checks if the name given exist as a db file,
+ * if not it gets created together with required tables. At initiation a file name is given making
+ * it possible to store a file per project depending how the classes that uses this handler
+ * implements this function.
+ *
+ */
+public class DataBaseHandler {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DataBaseHandler.class);
+    private static final String EVENT_ID_KEY = "eventId";
+    private final String databaseFile;
+
+    /**
+     * Constructor that takes plugin directory path as String and plugin name as String.
+     *
+     * @param pluginDir
+     * @param filename
+     * @throws ConnectException
+     */
+    public DataBaseHandler(final String pluginDir, final String filename) throws ConnectException {
+        Path filePath = Paths.get(pluginDir, filename);
+        this.databaseFile = String.format("jdbc:sqlite:%s", filePath);
+        createNewDatabase();
+        createTables();
+    }
+
+    /**
+     * Constructor that takes plugin directory path as File and plugin name as String.
+     *
+     * @param pluginDir
+     * @param filename
+     * @throws ConnectException
+     */
+    public DataBaseHandler(final File pluginDir, final String filename) throws ConnectException {
+        this(pluginDir.toString(), filename);
+    }
+
+    /**
+     * This sets up an enum for Tables including the key name used as primary key for that table
+     *
+     */
+    public enum Table {
```

Comment:
As this class is public it could be in its own file

Reply:
Done